backend/backend/activate_card.py

```diff
@@ -0,0 +1,109 @@
+""" Module to handle the activating of chance and community chest cards. """
+
+from random import randint
+from backend.player import Player
+from backend.game import Game
+from backend.properties import Property
+from backend.properties import get_properties
+from backend.cards import get_card_details
+
+LAST_CHANCE_INDEX_IN_TABLE = 14
+LAST_CHEST_INDEX_IN_TABLE = 29
+
+
+def activate_card(player_id, game_id, card_landed_on):
+    # pylint: disable=too-many-locals
+    """Read a chance or community chest card that the player landed on.
+
+    Arguments:
+        player_id: An int representing the current player id.
+        game_id: An int representing the current game id.
+        card_landed_on: An int representing the type of card the player has
+            activated
+
+    """
+    # Check if card is chance or community chest
+    if card_landed_on == "chance":
+        card_table_id = randint(0, LAST_CHANCE_INDEX_IN_TABLE)
+    else:
+        card_table_id = randint(LAST_CHANCE_INDEX_IN_TABLE + 1,
+                                LAST_CHEST_INDEX_IN_TABLE)
+
+    # Dive into database to get the card details
+    card_details = get_card_details(card_table_id)
+
+    # Check what type of chance card it is
+    card_type = card_details["operation_type"]
+
+    # Send the *type* and *description* to the client
+    # ...
+
+    # Get value
+    # This card_value is unique in that it represents different things
+    # depending on the card_type. See below comments for more information
+    card_value = card_details["value"]
+
+    # If it's a "move_specific" type of chance card
+    if card_type == "move_specific":
+        # Update the player position in players table with "chance" value
+        with Player(player_id) as player:
+            # Check if the card wants the player to move back some spaces
+            # or simply *jump* to a specific space on the board
+            if card_value < 0:
+                # The card value here indicates how many spaces to move back
+                player.board_position -= int(card_value)  # CHECK TYPES
```

Comment:
What does “check types” mean here? (Just curious)

Reply:
That was just a reminder for myself to check what the type of card_value was when it was retrieved from the database. Actually now that you ask, if there's an `int` in the database then will an `int` be returned in pymysql dictcursor, as per line 44 above?

Reply:
I’m not sure! I would *think* so, but I don’t know for certain.